app/src/app/search/search-filter-reducer.ts

```diff
@@ -8,39 +8,44 @@ export function searchFilterReducer(state: SearchState, action: SearchAction): S
           selected: f.selected.includes(value) ? f.selected.filter(id => id !== value) : [...f.selected, value]
         } : f
       )
-      return {...state, filters}
+      return {...state, filters, page: { ...state.page, value: 1 }}
     }
     case "toggle_radio_option": {
       const {name, value} = action.payload
       const filters = state.filters.map(f =>
         f.name === name ? {...f, selected: f.selected.find(id => id == value) ? [] : [value]} : f)
-      return {...state, filters}
+      return {...state, filters, page: { ...state.page, value: 1 }}
     }
     case "set_condition": {
       const {name, value, condition} = action.payload
       const filters = state.filters.map(f => f.name === name ? {...f, selected: [value], condition} : f)
-      return {...state, filters}
+      return {...state, filters, page: { ...state.page, value: 1 }}
     }
     case "set_search": {
       const {name, value} = action.payload
       const filters = state.filters.map(f => f.name === name ? {...f, selected: [value]} : f)
-      return {...state, filters}
+      return {...state, filters, page: { ...state.page, value: 1 }}
     }
     case "reset": {
       const {name} = action.payload
       const filters = state.filters.map(f => f.name === name ? {...f, selected: []} : f)
-      return {...state, filters}
+      return {...state, filters, page: { ...state.page, value: 1 }}
     }
     case "reset_all":
       const filters = state.filters.map(f => ({...f, selected: []}))
-      return {...state, filters}
+      return {...state, filters, page: { ...state.page, value: 1 }}
     case "set_order": {
       const {name} = action.payload
       const order = name == state.order.name
         ? {...state.order, direction: state.order.direction === "desc.nullslast" ? "asc" : "desc.nullslast"}
         : {name, direction: "desc.nullslast"}
       return {...state, order}
     }
+    case "set_page": {
+      const { value } = action.payload;
+      const page = { ...state.page, value }
+      return { ...state, page };
```

Comment:
For clarity, I suggest renaming `page` to`pagination` and also `page.value` to `pagination.pageNumber` or something similar.